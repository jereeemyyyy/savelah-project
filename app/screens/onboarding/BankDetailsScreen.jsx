import { View, Text, TextInput, TouchableOpacity } from 'react-native';
import { useState } from 'react';
import { useNavigation, useRoute } from "@react-navigation/native";
import BackButton from '../../components/BackButton';
import { supabase } from '../../../lib/supabase';

export default function BankDetailsScreen() {
    const [bankDetails, setBankDetails] = useState('');
    const navigation = useNavigation();
    const route = useRoute();
    const { username, userId } = route.params;

    const handleNext = async () => {
        // Call the function to save the username
        const { error } = await addBankDetails(userId, bankDetails);

        if (error) {
            console.error('Error saving username:', error);
        } else {
            navigation.navigate('Intro', { username, userId });
        }
    };

    const addBankDetails = async (userId, bankDetails) => {
        const { data, error } = await supabase
            .from('profiles')
            .update({ bankDetails })
            .eq('id', userId);

        return { data, error };
    };

    return (
        <View className="bg-gray-800 flex-1">
            <BackButton testID="backButton" />
            <Text testID="headerText" className="font-bold text-white text-4xl m-4">Enter Bank Account Details</Text>
            <Text testID="descriptionText" className="text-lg text-gray-400 mx-4 mb-4">Receive real time updates to update your expenses whenever you make a payment.</Text>

            <View className="flex-1 items-center m-5">
                <TextInput
                    testID="bankDetailsInput"
                    className="p-3 bg-gray-100 rounded mb-4 w-52"
                    value={bankDetails}
                    onChangeText={setBankDetails}
                    placeholder="Enter Bank Details"
                />
                <TouchableOpacity testID="nextButton" className="py-3 bg-indigo-500 rounded mb-4 w-20 items-center" onPress={handleNext}>
                    <Text className="text-white">Next</Text>
                </TouchableOpacity>
            </View>
        </View>
    );
}
