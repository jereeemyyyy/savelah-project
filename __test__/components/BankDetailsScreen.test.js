import React from 'react';
import renderer from 'react-test-renderer';
import BankDetailsScreen from '../../app/screens/onboarding/BankDetailsScreen';

jest.mock('@react-navigation/native', () => {
    const actualNav = jest.requireActual('@react-navigation/native');
    return {
        ...actualNav,
        useNavigation: () => ({
            navigate: jest.fn(),
        }),
        useRoute: () => ({
            params: {
                username: 'testuser',
                userId: '123',
            },
        }),
    };
});

describe('BankDetailsScreen', () => {
    it('renders correctly', () => {
        const tree = renderer.create(<BankDetailsScreen />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('contains the expected text elements', () => {
        const component = renderer.create(<BankDetailsScreen />);
        const instance = component.root;

        expect(instance.findByProps({ testID: 'headerText' }).props.children).toBe('Enter Bank Account Details');
        expect(instance.findByProps({ testID: 'descriptionText' }).props.children).toBe('Receive real time updates to update your expenses whenever you make a payment.');
    });

    it('contains the BackButton component', () => {
        const component = renderer.create(<BankDetailsScreen />);
        const instance = component.root;

        expect(instance.findByProps({testID: 'backButton'})).toBeTruthy();
    });

    it('contains the TextInput component', () => {
        const component = renderer.create(<BankDetailsScreen />);
        const instance = component.root;

        expect(instance.findByProps({testID: 'bankDetailsInput'})).toBeTruthy();
    });

    it('contains the TouchableOpacity component', () => {
        const component = renderer.create(<BankDetailsScreen />);
        const instance = component.root;

        expect(instance.findByProps({testID: 'nextButton'})).toBeTruthy();
    });
});
