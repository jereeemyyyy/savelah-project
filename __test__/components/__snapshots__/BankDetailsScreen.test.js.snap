// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BankDetailsScreen renders correctly 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "#1f2937",
      },
      {
        "flexBasis": "0%",
        "flexGrow": 1,
        "flexShrink": 1,
      },
    ]
  }
>
  <View
    style={
      [
        {
          "flexDirection": "row",
        },
        {
          "justifyContent": "flex-start",
        },
      ]
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "backgroundColor": "#6366f1",
          "borderBottomLeftRadius": 16,
          "borderTopRightRadius": 16,
          "marginLeft": 16,
          "opacity": 1,
          "paddingBottom": 8,
          "paddingLeft": 8,
          "paddingRight": 8,
          "paddingTop": 8,
        }
      }
    >
      <RNSVGSvgView
        align="xMidYMid"
        bbHeight="20"
        bbWidth="20"
        color="white"
        data-slot="icon"
        fill="currentColor"
        focusable={false}
        height="20"
        meetOrSlice={0}
        minX={0}
        minY={0}
        style={
          [
            {
              "backgroundColor": "transparent",
              "borderWidth": 0,
            },
            {
              "flex": 0,
              "height": 20,
              "width": 20,
            },
          ]
        }
        tintColor="white"
        vbHeight={24}
        vbWidth={24}
        width="20"
      >
        <RNSVGGroup
          fill={
            {
              "type": 2,
            }
          }
          propList={
            [
              "fill",
            ]
          }
        >
          <RNSVGPath
            clipRule={0}
            d="M11.03 3.97a.75.75 0 0 1 0 1.06l-6.22 6.22H21a.75.75 0 0 1 0 1.5H4.81l6.22 6.22a.75.75 0 1 1-1.06 1.06l-7.5-7.5a.75.75 0 0 1 0-1.06l7.5-7.5a.75.75 0 0 1 1.06 0Z"
            fill={
              {
                "payload": 4278190080,
                "type": 0,
              }
            }
            fillRule={0}
            propList={
              [
                "fillRule",
              ]
            }
          />
        </RNSVGGroup>
      </RNSVGSvgView>
    </View>
  </View>
  <Text
    style={
      [
        {
          "fontWeight": "700",
        },
        {
          "color": "#fff",
        },
        {
          "fontSize": 36,
          "lineHeight": 40,
        },
        {
          "marginBottom": 16,
          "marginLeft": 16,
          "marginRight": 16,
          "marginTop": 16,
        },
      ]
    }
    testID="headerText"
  >
    Enter Bank Account Details
  </Text>
  <Text
    style={
      [
        {
          "fontSize": 18,
          "lineHeight": 28,
        },
        {
          "color": "#9ca3af",
        },
        {
          "marginLeft": 16,
          "marginRight": 16,
        },
        {
          "marginBottom": 16,
        },
      ]
    }
    testID="descriptionText"
  >
    Receive real time updates to update your expenses whenever you make a payment.
  </Text>
  <View
    style={
      [
        {
          "flexBasis": "0%",
          "flexGrow": 1,
          "flexShrink": 1,
        },
        {
          "alignItems": "center",
        },
        {
          "marginBottom": 20,
          "marginLeft": 20,
          "marginRight": 20,
          "marginTop": 20,
        },
      ]
    }
  >
    <TextInput
      onChangeText={[Function]}
      placeholder="Enter Bank Details"
      style={
        [
          {
            "paddingBottom": 12,
            "paddingLeft": 12,
            "paddingRight": 12,
            "paddingTop": 12,
          },
          {
            "backgroundColor": "#f3f4f6",
          },
          {
            "borderBottomLeftRadius": 4,
            "borderBottomRightRadius": 4,
            "borderTopLeftRadius": 4,
            "borderTopRightRadius": 4,
          },
          {
            "marginBottom": 16,
          },
          {
            "width": 208,
          },
        ]
      }
      testID="bankDetailsInput"
      value=""
    />
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#6366f1",
          "borderBottomLeftRadius": 4,
          "borderBottomRightRadius": 4,
          "borderTopLeftRadius": 4,
          "borderTopRightRadius": 4,
          "marginBottom": 16,
          "opacity": 1,
          "paddingBottom": 12,
          "paddingTop": 12,
          "width": 80,
        }
      }
      testID="nextButton"
    >
      <Text
        style={
          [
            {
              "color": "#fff",
            },
          ]
        }
      >
        Next
      </Text>
    </View>
  </View>
</View>
`;
